xclim/subset.py

```diff
@@ -164,6 +165,37 @@ def func_checker(*args, **kwargs):
     return func_checker
 
 
+def check_latlon_dimnames(func):
+    @wraps(func)
+    def func_checker(*args, **kwargs):
+        """
+        A decorator examining the names of the latitude and longitude dimensions and renames them temporarily if needed.
+        """
+
+        if range(len(args)) == 0:
+            return func(*args, **kwargs)
+
+        formatted_args = list()
+        for argument in args:
+            try:
+                if isinstance(argument, (xarray.DataArray, xarray.Dataset)):
+                    dims = argument.dims
+                else:
+                    raise TypeError
+            except (IndexError, TypeError):
+                logging.error(f"No file or no dimensions found in arg {argument}.")
+                formatted_args.append(argument)
+                continue
+
+            if not {"lon", "lat"}.issubset(dims):
+                if {"latitude", "longitude"}.issubset(dims):
+                    argument = argument.rename({"latitude": "lat", "longitude": "lon"})
+            formatted_args.append(argument)
+        return func(*formatted_args, **kwargs)
```

Comment:
Question: should we convert the output dimension names back to their original values ? 
Also, what if the name is "long" instead of "lon" ?

Reply:
I feel like it would be good to be consistent with inputs and outputs, even if they aren't *exactly* using the standard. I was thinking that a way of implementing it could be using a dictionary to keep track of name changes and, if that dictionary isn't empty, call the function within the decorator and perform a dimension rename on exit.

Reply:
Just thinking here, but another way to go about this is to write a function `to_cf` that would try to convert a dataset to the CF convention. 
What I'm concerned here is that with this decorator, the subset function has an undocumented and unexpected side-effect (dimension name change).

Reply:
`long` is absolutely possible and an edge case that could go in here.

Reply:
The alternative is to figure out what the longitude and latitude dimensions are and tell all subsetting functions to use those instead of `lon` and `lat`. That might be a lot more work.

Reply:
This is why I think it would be good to not change the output dimension names from what they were at the start. I do like the `to_cf` method you're thinking of though. Could we push some of this code to a function like that?

Reply:
I think so. And you know what would be nice ? If the `to_cf` method could also output a NcML file embedding this conversion from original file to cf-compliant.

Reply:
Just an idea here, I know nothing about CF conventions so it might sound noob or impossible.

Are we allowed to create alias/alternate name/symlink to the old name that is not good?  So there are no "unexpected side-effect (dimension name change)", we are just adding an alias to make it compliant.

Reply:
Honestly, the idea behind this is simply to make it easier to perform operations on those coordinates. An alias could certainly work for that.

@huard The idea of getting things closer to CF-conformance is definitely something to bring into xclim. I think we've hinted at it in a few issues. Might be good to open a new one with the laundry list of what we would want (similar to the data quality assurance checks discussion).

Reply:
The alias idea is not crazy, but this is something that would go into xarray. Given that renames are simple operations, I'm not convinced this would fly.

Reply:
see #350 for the follow-up discussion on this. We can eventually port some of this code to a better method once we figure out how best to proceed.